### PR TITLE
docs: Make contributing docs more beginner-oriented

### PR DESCRIPTION
**Pull request by drwpow**
## Changes
Reading the contributor docs, it assumed someone had a commanding knowledge of Git and open source contribution. Made some tweaks to the language to offer just a little more explanation without being too wordy, in the hopes that they’ll be a more helpful beginner guide (after all, what open source expert meticulously reads every word in the “how to make a pull request“ section of contributing docs?).

Read doc here: https://github.com/snowpackjs/astro/blob/contribuing-docs/CONTRIBUTING.md

## Testing

Is docs

## Docs
Is docs

### CODE REVIEW

File: CONTRIBUTING.md [hunk 1]
```diff
@@ -13,53 +13,114 @@ At the same time, I had a day job where I was a junior software developer at a r
...
 - **Learning from smart people:** I got to meet and learn from so many smart people across the open source ecosystem.
-- **preventing imposter syndrome:** Sure, I was still just a kid, but having an actual human connection to developers who I looked up to at the time helped dispell the idea that "oh, **I** could never be like that."
+- **preventing imposter syndrome:** Sure, I was still just a kid, but having an actual human connection to developers who I looked up to at the time helped dispell the idea that "oh, _I_ could never be like that."
 - **Making friends in the larger community:** The creator of request, [@mikeal](https://twitter.com/mikeal), is still a friend to this day.
 
-If any of this sounds interesting, I hope you consider getting involved with Astro. Come say hi in the [**#new-contributors**](https://astro.build/chat) channel on Discord, anytime. We're always around and value contributions of any shape/size.
+If any of this sounds interesting, I hope you consider getting involved with Astro. Come say hi in the [**#new-contributors**][discord] channel on Discord anytime. We're always around and value contributions of any shape/size.
```
Comment (FredKSchott) [inline CONTRIBUTING.md:22]: conversation in #1120 led to the removal of this for now. Will probably re-post the meat of this to a personal blog and then add back a pointer to that blog.

In that case, I'd recommend removing these changes to prevent merge conflicts later.

File: CONTRIBUTING.md [hunk 2]
```diff
@@ -13,53 +13,114 @@ At the same time, I had a day job where I was a junior software developer at a r
...
+- [Git 2.x][install-git]
 
-Astro uses yarn workspaces, so you should **always run `yarn install` from the top-level project directory.** running `yarn install` in the top-level project root will install dependencies for `astro`, `www`, `docs`, and every package in the repo.
+## Setting up Astro locally
 
-```shell
-git clone && cd ...
-yarn install
-yarn build:all
+All work on Astro begins by downloading a copy locally. Start by [**forking**][how-to-fork] this repository (if you’re new to this, [GitHub has a great detailed guide][how-to-fork]). You should end up with an `astro/` code folder on your local machine.
```
Comment (FredKSchott) [inline CONTRIBUTING.md:36]: "`astro/` code folder" sounds like some special thing, rephrase to "cloned `astro/` folder"?

File: CONTRIBUTING.md [hunk 3]
```diff
@@ -13,53 +13,114 @@ At the same time, I had a day job where I was a junior software developer at a r
...
+
+Switch to this folder by opening your **Terminal application** of choice and running:
```
Comment (FredKSchott) [inline CONTRIBUTING.md:38]: wouldn't the user already have their terminal open to clone the repo? This confused me a bit, so maybe add a (if you haven't opened it already).

File: CONTRIBUTING.md [hunk 4]
```diff
@@ -13,53 +13,114 @@ At the same time, I had a day job where I was a junior software developer at a r
...
+
+```
+cd /[path to my code folder]/astro
```
Comment (FredKSchott) [inline CONTRIBUTING.md:41]: is this confusing for people who did use the Terminal to clone, in which case it's `cd astro` instead of `cd /.../astro`? Confused me a little bit

File: CONTRIBUTING.md [hunk 5]
```diff
@@ -13,53 +13,114 @@ At the same time, I had a day job where I was a junior software developer at a r
...
 ```
 
+Next, you’ll need to install dependencies and set up the project. You can do that by running the following commands from within the `astro/` folder:
```
Comment (FredKSchott) [inline CONTRIBUTING.md:44]: Looks like all references have been removed so many a non-issue, but I liked the idea of defining "top-level project root" as a concept / term here so that we could use it later in the document.

"`astro/` folder" is confusing since it could also refer to the `astro/` folder inside of `packages/`.

File: CONTRIBUTING.md [hunk 6]
```diff
@@ -13,53 +13,114 @@ At the same time, I had a day job where I was a junior software developer at a r
...
+yarn build
+```
+
+This will download everything you need and build Astro locally for use. You’re ready to go!
+
 ## Development
 
-```shell
-# starts a file-watching, live-reloading dev script for active development
+While working within packages (such as `packages/astro/*`), you may notice you have to run `yarn build` to see your changes. This process can really slow you down if you are trying many changes at once. Try running the following instead:
```
Comment (FredKSchott) [inline CONTRIBUTING.md:55]: `you may notice you have to` feels like its contradicted by the rest of this. How about:

```suggestion
Running `yarn build` after every change can really slow you down, especially when you are making many changes at once. During development, try running the following command instead:
```

File: CONTRIBUTING.md [hunk 7]
```diff
@@ -13,53 +13,114 @@ At the same time, I had a day job where I was a junior software developer at a r
...
+
+```
 yarn dev
-# build the entire project, one time.
+```
+
+This will start a build watcher, which will rebuild files as you work. Use this to speed up your workflow, and see your changes more rapidly.
+
+Of course, you can still build all files within a package, as before. You may need to do so from time to time:
```
Comment (FredKSchott) [inline CONTRIBUTING.md:63]: Q: when might I need to do so?

File: CONTRIBUTING.md [hunk 8]
```diff
@@ -13,53 +13,114 @@ At the same time, I had a day job where I was a junior software developer at a r
...
 ```
 
 ## Running tests
 
-```shell
-# run this in the top-level project root to run all tests
+You’ll need to run existing tests to make sure your work didn’t cause any regressions. To create a fresh build and run all tests, run the following from the project root:
+
+```
+yarn
```
Comment (FredKSchott) [inline CONTRIBUTING.md:74]: nit: `yarn` only needs to be run once, and `yarn build` isn't needed when `yarn dev` is already running. Is there a way to communicate this so that users aren't running 3 commands every time they test?

File: CONTRIBUTING.md [hunk 9]
```diff
@@ -13,53 +13,114 @@ At the same time, I had a day job where I was a junior software developer at a r
...
-# run only a few tests, great for working on a single feature
-# (example - `yarn test rss` runs `astro-rss.test.js` tests)
-yarn test $STRING_MATCH
 ```
 
+If you find yourself wanting to not run the whole test suite, you can run the following instead:
+
+```
+cd packages/astro
+yarn test astro-markdown
```
Comment (FredKSchott) [inline CONTRIBUTING.md:83]: this actually works from the top level! you can remove `cd packages/astro`.

Comment (FredKSchott) [reply CONTRIBUTING.md:83]: also personal preference, but I feel like the existing snippet communicated the same info more clearly. Just a thought in case you're feeling the same!

File: CONTRIBUTING.md [hunk 10]
```diff
@@ -13,53 +13,114 @@ At the same time, I had a day job where I was a junior software developer at a r
...
+```
+
+Replace “`astro-markdown`“ with any part of a test filename to only run those tests.
+
+## Making a Pull Request
```
Comment (FredKSchott) [inline CONTRIBUTING.md:88]: some questions:

1. Most developers can't push directly to `origin`, so this needs to be updated to describe how Git/Github forks work.
2. Asking more broadly: Can we link to some good GitHub docs on this topic instead? I feel like this is a bit out of scope for our contributor docs.
3. Asking even more broadly: Are we able to take on contributions from people who aren't familiar with how GitHub works? I think the answer is yes, but I just want to be sure.

my 2 cents: I actually like that a developer doesn't currently need to worry about changesets until _after_ they have already submitted the PR and get the warning from the bot. It reduces the time to "I did it! success!" The changeset work becomes its own small followup task, with no impact on the already steep learning curve. Bonus: the bot documents how to do it, maintains that documentation, etc. so that we don't have to.

The other comments are more nits and personal preference, but this one I do feel a bit more strongly about.

File: CONTRIBUTING.md [hunk 11]
```diff
@@ -13,53 +13,114 @@ At the same time, I had a day job where I was a junior software developer at a r
...
+
+   Also, while Astro is pre-1.0, you probably want to only make **patch** changes, unless you’re releasing a huge, sweeping, breaking change as a **minor** release ([what is semver?][semver]).
+
+1. [Open a new Pull Request][astro-pr] on Astro with your changes.
+
+Be sure to **fill in the template** completely! It helps other people review your work.
+
+If you need to create a pull request to get early feedback, that’s OK, too.
+
+Creating your first Pull Request can seem like a lot of steps at first, but with time and practice it’ll become comfortable.
```
Comment (FredKSchott) [inline CONTRIBUTING.md:122]: fyi this is what made me think of the "my 2 cents" comment above. If we reduced this section to just: "Here's a great GitHub-written and reviewed guide on how to submit your first Pull Request" I think we would get more contributions and a lot fewer "hey, I did X from the docs and it didn't work" support requests.

**Review by FredKSchott — COMMENTED** [review]
Love this focus for the contributing docs!

File: CONTRIBUTING.md (hunk 3 above)
Comment (drwpow) [reply CONTRIBUTING.md:38]: Not necessarily; many users will use the GitHub Desktop client or some other assistive software to do so.

File: CONTRIBUTING.md (hunk 8 above)
Comment (drwpow) [reply CONTRIBUTING.md:74]: > `yarn build` isn't needed when `yarn dev` is already running. 

There’s actually a lot of cases where this isn’t true. 2 main scenarios are if files were created, deleted, or renamed, `yarn build` needs to be run. Also, there’s a bug in our dev scripts where some post-build actions are skipped, resulting in errors (e.g. the `postbuild` script in `astro-markdown`). For beginners, I think this is confusing, so I just recommend running a fresh (correct) build before running tests.

File: CONTRIBUTING.md (hunk 9 above)
Comment (drwpow) [reply CONTRIBUTING.md:83]: I’ll double-check; this used to not work from the top level

Comment (drwpow) [reply CONTRIBUTING.md:83]: Ah this was fixed! OK will revert the original.

File: CONTRIBUTING.md (hunk 10 above)
Comment (drwpow) [reply CONTRIBUTING.md:88]: Didn’t the document above link to a guide on forking and start from the standpoint of them working from their own fork?

File: CONTRIBUTING.md (hunk 3 above)
Comment (FredKSchott) [reply CONTRIBUTING.md:38]: I'm +1 for anything that accounts for both types of users

File: CONTRIBUTING.md (hunk 8 above)
Comment (FredKSchott) [reply CONTRIBUTING.md:74]: Totally agreed if its been a while, but my fear is that the docs seem to suggest this flow:

- make a change
- run `yarn`
- run `yarn build`
- run `yarn test`
- make another change
- run `yarn`
- run `yarn build`
- etc. etc.

Can you communicate what you want to communicate, in a way that just makes it really clear that `yarn` and `yarn build` aren't needed every time?

File: CONTRIBUTING.md (hunk 10 above)
Comment (FredKSchott) [reply CONTRIBUTING.md:88]: oh right, `origin` in this case is their fork and not ours. nvm then on num 1! The rest of the comments/questions still stand.

File: CONTRIBUTING.md (hunk 8 above)
Comment (drwpow) [reply CONTRIBUTING.md:74]: Perhaps I’ll just add a section outlining all the scripts separately and reference back to them.

**Comment by drwpow** [comment]
Closing for now; I don’t think I’ll have time to get to these revisions soon, and they may conflict anyway. Will revisit when I have more time to address everything.

**Comment by FredKSchott** [comment]
Didn't mean to scare you away with my review! If there are smaller changes that you can get in, feel free.
